packages/backend/src/apps/m365-excel/actions/update-table-row/index.ts

```diff
@@ -99,6 +100,16 @@ const action: IRawAction = {
     const { fileId, tableId, lookupColumn, lookupValue, columnsToUpdate } =
       parametersParseResult.data
 
+    // Validation to prevent path traversals
+    if (!TABLE_ID_WITH_BRACES_REGEX.test(String(tableId))) {
+      throw new StepError(
+        'Table is of an invalid format',
+        'Check that your table is selected correctly.',
+        $.step.position,
+        $.app.name,
+      )
+    }
```

Comment:
move this to separate function to validate table id and file id or else this step error is re-typed in 3 places.